app/src/main/java/com/chess/clock/manager/ChessClockManagerImpl.java

```diff
@@ -0,0 +1,192 @@
+        mPlayerOneTimer.resume();
+        mPlayerTwoTimer.resume();
+        log("Resumed the clock timers.");
+    }
+
+    @Override
+    public void resetClock() {
+        mPlayerOneTimer.resetTimeControl();
+        mPlayerTwoTimer.resetTimeControl();
+        chessGameRunning = false;
+    }
+
+    @Override
+    public long getTimeForPlayer(ClockPlayer player) {
+        long time = 0L;
+        switch (player) {
+            case ONE:
+                time = mPlayerOneTimer.getTime();
+                break;
+            case TWO:
+                time = mPlayerTwoTimer.getTime();
+                break;
+        }
+        return time;
+    }
+
+    @Override
+    public void setPlayerTime(ClockPlayer player, long timeMs) {
+        switch (player) {
+            case ONE:
+                mPlayerOneTimer.setTime(timeMs);
+                break;
+            case TWO:
+                mPlayerTwoTimer.setTime(timeMs);
+                break;
+        }
+    }
+
+
+    /**
+     * Notifies the Chess Clock Service that a player made a move.
+     *
+     * @param playerTimer   The player timer which made a move.
+     * @param opponentTimer The opponent timer.
+     */
+    private void pressPlayerClock(CountDownTimer playerTimer, CountDownTimer opponentTimer) {
+
+        // Ignore clock press if timers are not initiated or opponent timer already finished.
+        if (!opponentTimer.isFinished() && !playerTimer.isFinished()) {
+
+            // Game already running? stop player timer and start the opponent timer.
+            if (chessGameRunning) {
+                playerTimer.stop();
+                opponentTimer.start();
+
+                log("Move number: " + playerTimer.getTotalMoveCount() + ", time left: " + ClockTime.raw(playerTimer.getTime()).toReadableFormat());
+            }
+            // First move: do not stop player clock to avoid invalid initial time increment.
+            else {
+                opponentTimer.start();
+                chessGameRunning = true;
+
+                log("Game started.");
+            }
+        } else {
+            log("Discarded clock press due to Time Controls not available or game finished already.");
+        }
+    }
+
+    @Override
+    public boolean isClockStarted() {
+        return mPlayerOneTimer.isStarted() || mPlayerTwoTimer.isStarted();
+    }
+
+    @Override
+    public void setListeners(CountDownTimer.Callback playerOneCallback, CountDownTimer.Callback playerTwoCallback) {
+        if (playerOneCallback != null) {
+            log("(1)registered listener: #" + playerOneCallback.hashCode());
+        }
+        mPlayerOneTimer.setClockTimerListener(playerOneCallback);
+
+        if (playerTwoCallback != null) {
+            log("(2) registered listener: #" + playerTwoCallback.hashCode());
+        }
+        mPlayerTwoTimer.setClockTimerListener(playerTwoCallback);
+    }
+
+    private void log(String message) {
+        if (!BuildConfig.DEBUG) return;
```

Comment:
Nit: this probably doesn't matter, because proguard should remove logs from release build anyway https://github.com/ChessCom/android-chessclock/blob/master/app/proguard-rules.pro#L4
Of course if `message` creation is expensive that will still happen, but that's still the case with manual `BuildConfig.DEBUG` check.